cmd/podman/create.go

```diff
@@ -169,21 +170,41 @@ func createContainer(c *cliconfig.PodmanCommand, runtime *libpod.Runtime) (*libp
 			imageName = newImage.ID()
 		}
 
-		// add healthcheck if it exists AND is correct mediatype
-		_, mediaType, err := newImage.Manifest(ctx)
-		if err != nil {
-			return nil, nil, errors.Wrapf(err, "unable to determine mediatype of image %s", newImage.ID())
+		var healthCheckCommandInput string
+		// if the user disabled the healthcheck with "none", we skip adding it
+		if c.Flags().Lookup("healthcheck-command") != nil {
+			healthCheckCommandInput = c.String("healthcheck-command")
 		}
-		if mediaType == manifest.DockerV2Schema2MediaType {
-			healthCheck, err = newImage.GetHealthCheck(ctx)
+
+		// the user didnt disable the healthcheck but did pass in a healthcheck command
+		// now we need to make a healthcheck from the commandline input
+		if healthCheckCommandInput != "none" && len(healthCheckCommandInput) > 0 {
+			healthCheck, err = makeHealthCheckFromCli(c)
 			if err != nil {
-				return nil, nil, errors.Wrapf(err, "unable to get healthcheck for %s", c.InputArgs[0])
+				return nil, nil, errors.Wrapf(err, "unable to create healthcheck")
 			}
 			if healthCheck != nil {
 				hasHealthCheck = true
 			}
+		} else if healthCheckCommandInput != "none" && len(healthCheckCommandInput) == 0 {
+			// the user did not disable the health check and did not pass in a healthcheck
+			// command as input.  so now we add healthcheck if it exists AND is correct mediatype
+			_, mediaType, err := newImage.Manifest(ctx)
+			if err != nil {
+				return nil, nil, errors.Wrapf(err, "unable to determine mediatype of image %s", newImage.ID())
+			}
+			if mediaType == manifest.DockerV2Schema2MediaType {
```

Comment:
Do we need to do this check?  

If I have a nonDockerV2Schema2MediaType image, won't newImage,GetHealthCheck(ctx) just return ""?

That way if OCI Images or some future image got HealthCheck support this code would not need to change.

Reply:
if it isnt a big deal to you, id like to leave as is for a couple of reasons:
1. i dont think any other image would implement this in the same type (manifest.Schema2HealthConfig)
2. i want to avoid unexpected nil errors

if you feel strongly, ill provide a rework.